runtimes/rust/src/lib.rs

```diff
@@ -635,8 +635,10 @@ impl Runtime for RustRuntime {
             }
             InstanceStatus::Migrating(_) => { /* Do nothing. */ }
             other => {
-                log::warn!(
-                    "Received non-expected service status: {}; ignoring the request",
+                panic!(
+                    "Received non-expected service status: {}; \
+                     Rust runtime isn't prepared to process this action, \
+                     probably node binary is configured incorrectly",
```

Comment:
Nit: I'd specify that the rust runtime is probably outdated relative to the core library in this case; it could help pinpoint the problem. Also, I wonder if it could make sense to have a method `InstanceStatus::can_be_ignored()` for runtimes, which would allow to not panic here if a status change is not "significant" (i.e., can be ignored by a typical runtime).